src/main/java/googlesignin/GoogleUser.java

```diff
@@ -0,0 +1,88 @@
+package googlesignin;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleTokenResponse;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.people.v1.PeopleService;
+import com.google.api.services.people.v1.model.Person;
+import com.google.api.services.people.v1.model.PhoneNumber;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import dbaccessor.user.UserAccessor;
+import entity.User;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+public final class GoogleUser {
+
+    @Autowired
+    private UserAccessor userAccessor;
+
+    private final GoogleTokenResponse tokenResponse;
+
+    public GoogleUser(GoogleTokenResponse tokenResponse) {
+        this.tokenResponse = tokenResponse;
+    }
+
+    private GoogleCredentials getCredentials() throws IOException {
+        String accessTokenString = tokenResponse.getAccessToken();
+        AccessToken accessToken = new AccessToken(accessTokenString, null);
+        return new GoogleCredentials(accessToken);
+    }
+
+    public Optional<String> getPhoneNumber(String userId) throws IOException {
+        HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(getCredentials());
+        HttpTransport httpTransport = new NetHttpTransport();
+        JacksonFactory jsonFactory = new JacksonFactory();
+        PeopleService peopleService =
+                new PeopleService.Builder(httpTransport, jsonFactory, requestInitializer).build();
+        Person profile = peopleService.people().get("people/" + userId)
+                .setPersonFields("phoneNumbers")
+                .execute();
+        List<PhoneNumber> phoneNumberList = profile.getPhoneNumbers();
```

Comment:
Try to use Immutable list if possible. Here and elsewhere

Reply:
done.